### PR TITLE
fix: typos in CollateralTracker

### PR DESCRIPTION
**Pull request by 0xScratch**
Three types of changes made:

**1. Correct and Optimized packing of storage variables**: This is well explained here -> https://github.com/allo-protocol/allo-v2/pull/237

**2. Wastage of Gas in Initialization**: Usually all unsigned integers have a default value of 0, that means if I included this code line -> `uint256 num;`, this initialized num with 0. Thus, changes in code are made from `uint256 leg = 0;` to `uint256 leg;`

**3. Some Grammatical and Typo fixes**

### CODE REVIEW

File: contracts/CollateralTracker.sol [hunk 1]
```diff
@@ -134,15 +134,15 @@ contract CollateralTracker is ERC20Minimal, Multicall {
     /// @notice a token can only be initialized once - which is done by the Panoptic pool.
     bool internal s_initialized;
 
+    /// @dev store whether the current collateral token is token0 of the AMM (true) or token1 (false).
```
Comment (dyedm1) [inline contracts/CollateralTracker.sol]: Please revert this change. The `bool` type is packed as 1 bit in storage, so it will get packed with `s_univ3Token1` and take up the same number of storage slots as what you have changed it to. 
Current: <160 + 1> <160><160+1>|<160>...
Yours: <160 +1 +1><160><160>|<160>...

File: contracts/CollateralTracker.sol [hunk 2]
```diff
@@ -228,7 +228,7 @@ contract CollateralTracker is ERC20Minimal, Multicall {
     /// @notice Initialize a new collateral tracker for a specific token corresponding to the Panoptic Pool being created by the factory that called it.
     /// @dev The factory calls this function to start a new collateral tracking system for the incoming token at 'underlyingToken'.
     /// The factory will do this for each of the two tokens being tracked. Thus, the collateral tracking system does not track *both* tokens at once.
-    /// @param underlyingToken the address of the token to track collateral for corresponding to the Panoptic Pool being created by the factory calling this (as 'msg.sender').
```
Comment (dyedm1) [inline contracts/CollateralTracker.sol]: Please revert this, the current grammar is as intended. Without the "for", it is unclear that we are tracking collateral balances *specifically* of that token

File: contracts/CollateralTracker.sol [hunk 3]
```diff
@@ -818,7 +818,7 @@ contract CollateralTracker is ERC20Minimal, Multicall {
 
     /// @notice Get the cost of exercising an option. Used during a forced exercise.
     /// @dev This one computes the cost of calling the forceExercise function on a position:
-    /// - The forceExercisor will have to *pay* the exercisee because their position will be closed "against their will"
```
Comment (dyedm1) [inline contracts/CollateralTracker.sol]: Please revert, exercise is not gramatically correct here. exercisee=user who is being force exercised

File: contracts/CollateralTracker.sol [hunk 4]
```diff
@@ -850,7 +850,7 @@ contract CollateralTracker is ERC20Minimal, Multicall {
         uint160 currentSqrtPriceX96 = TickMath.getSqrtRatioAtTick(currentTick);
         uint160 medianSqrtPriceX96 = TickMath.getSqrtRatioAtTick(medianTick);
         unchecked {
-            for (uint256 leg = 0; leg < TokenId.countLegs(positionId); ++leg) {
+            for (uint256 leg; leg < TokenId.countLegs(positionId); ++leg) {
```
Comment (dyedm1) [inline contracts/CollateralTracker.sol]: Please revert. Although this is logically identical, we explicitly specify =0 here for clarity.

File: contracts/CollateralTracker.sol [hunk 5]
```diff
@@ -945,7 +945,7 @@ contract CollateralTracker is ERC20Minimal, Multicall {
 
     /// @notice Returns the original delegated value to a user at a certain tick based on the available collateral from the exercised user.
     /// @dev Only called on collateralTracker0, so we must query balances from collateralTracker1.
-    /// @param refunder Address of the user the refund is coming from (the force exercisee).
+    /// @param refunder Address of the user the refund is coming from (the force exercise).
```
Comment (dyedm1) [inline contracts/CollateralTracker.sol]: Please revert, exercise is not gramatically correct here. exercisee=user who is being force exercised

File: contracts/CollateralTracker.sol [hunk 6]
```diff
@@ -1495,7 +1495,7 @@ contract CollateralTracker is ERC20Minimal, Multicall {
         // Offset determined whether to consider the last tokenId from the list
         // (a potentially newly minted position)
         uint256 totalIterations = positionBalanceArray.length;
-        for (uint256 i = 0; i < totalIterations; ) {
```
Comment (dyedm1) [inline contracts/CollateralTracker.sol]: Please revert. Although this is logically identical, we explicitly specify =0 here for clarity.

File: contracts/CollateralTracker.sol [hunk 7]
```diff
@@ -1542,7 +1542,7 @@ contract CollateralTracker is ERC20Minimal, Multicall {
         uint256 numLegs = tokenId.countLegs();
 
         unchecked {
-            for (uint256 index = 0; index < numLegs; ++index) {
+            for (uint256 index; index < numLegs; ++index) {
```
Comment (dyedm1) [inline contracts/CollateralTracker.sol]: Please revert. Although this is logically identical, we explicitly specify =0 here for clarity.

**Review by dyedm1 — CHANGES_REQUESTED** [review]
Thanks for the PR! I'm going to have ask you to revert most of the code changes, but there's still some valid spelling fixes :)

**Comment by 0xScratch** [comment]
Done!!

**Review by dyedm1 — APPROVED** [review]
LGTM!
